contracts/token-distribution/SaleRounds.sol

```diff
@@ -92,166 +114,174 @@ contract SaleRounds is TokenDistribution, GameOwner, ERC20 {
-            revert("User has already registered for different round");
-        }
+        return reservedBalances[roundType][_to];
+    }
 
-        if(reservedBalances[RoundType.PRIVATE][_to] > 0 && roundType != RoundType.PRIVATE){
-            revert("User has already registered for different round");
-        }
+    function getCliffTime(string calldata _roundType) external view onlyGameOwner returns(uint256) {
+        RoundType roundType =  getRoundTypeByKey(_roundType);
 
-        reserveTokensInternal(roundType, _to, _amount);
+        return roundDistribution[roundType].cliff;
     }
 
     // @_amount is going be decimals() == default(18) digits
-    function mintTokensForPublic(string calldata _roundType, address _to, uint _amount) public
-        onlyOwner isRoundActive(_roundType) {
-        RoundType roundType = getRoundTypeByKey(_roundType);
-
-        require(roundType == RoundType.PUBLIC , "round type is not valid");
-        require(roundDistribution[roundType].totalRemaining >= _amount, "total remaining amount is not enough" );
+    function mintTokensForPublic(address _to, uint _amount) private
+    onlyGameOwner {
+        require(roundDistribution[RoundType.PUBLIC].totalRemaining >= _amount, "total remaining amount is not enough");
 
-        roundDistribution[roundType].totalRemaining -= _amount;
+        roundDistribution[RoundType.PUBLIC].totalRemaining -= _amount;
         _mint(_to, _amount);
     }
 
-    function claimTokens(string calldata _roundType, address _to) public
-        isRoundActive(_roundType) claimableRound(_roundType) {
-        RoundType roundType = getRoundTypeByKey(_roundType);
-        require(_msgSender() == _to, "Sender is not a recipient");
+    // @_amount is going be decimals() == default(18) digits
+    function reserveTokensInternal(RoundType _roundType, address _to, uint _amount) private {
+        require(roundDistribution[_roundType].supply >= _amount, "given amount is bigger than max supply for the round");
+        require(roundDistribution[_roundType].totalRemaining >= _amount, "total remaining round amount is not enough");
+        require(_to != address(0), "Reservation address is 0x0 address");
+        roundDistribution[_roundType].totalRemaining -= _amount;
+        roundDistribution[_roundType].startTime = block.timestamp;
+        reservedBalances[_roundType][_to] += _amount;
+    }
 
-        ClaimInfo memory claimInfo = ClaimInfo({cliff: roundDistribution[roundType].cliff,
-                                                vesting: roundDistribution[roundType].vesting,
-                                                balance: reservedBalances[roundType][_to],
-                                                claimedBalance: claimedBalances[roundType][_to],
-                                                //granularity can be 30 * 24 * 60 * 60 for monthly or 24 * 60 * 60 for daily? add a field and make constants
```

Comment:
Vestigial comment, granularity has already been addressed as constants.

Reply:
This was deleted already.